lib/device.js

```diff
@@ -251,6 +256,168 @@ const util = require('util'),
                 }
                 return returnTxt;
             }
+        },
+        /**
+         * get screen capture of the given device
+         * @property options {String} device, display, outputPath
+         */
+        captureScreen: function(options, next) {
+            log.info("device#capture#captureScreen()");
+            if (typeof next !== 'function') {
+                throw new Error('Missing completion callback (next=' + util.inspect(next) + ')');
+            }
+            options = options || {};
+            async.series([
+                _makeSession,
+                _makeCaptureOption,
+                _captureFile,
+                _copyFileToLocal
+            ],  function(err, results) {
+                log.verbose("device#capture()", "err: ", err, "results:", results);
+                const resultTxt = "Create " + chalk.green(options.captureFileName) + " to " + options.destinationPath +"\nSuccess";
+
+                // clean up /tmp/aresCapture directory in target device
+                if (options.createdTmpDir) {
+                    _removeTmpDir(function finish(removeErr) {
+                        next(err ? err : removeErr, resultTxt);
+                    });
+                } else {
+                    next(err, resultTxt);
+                }
+            });
+
+            function _makeSession(next) {
+                options.nReplies = 1; // -n 1
+                options.session = new novacom.Session(options.device, next);
+            }
+
+            function _makeCaptureOption(next) {
+                log.info("device#capture#_makeCaptureOption()");
+                let captureFileName = "",
+                    destinationPath = "",
+                    captureFormat = "PNG"; // PNG is default format
+
+                if (options.outputPath === null){
+                    captureFileName = _createDateFileName();
+                    destinationPath = path.resolve('.');
+                } else {
+                    // Get directory path, file path, file Extenstion from given path
+                    const parseDirPath = path.dirname(options.outputPath),
+                        parseBase = path.parse(options.outputPath).base;
+
+                    let parseExt = path.parse(options.outputPath).ext;
+                    parseExt = parseExt.split('.').pop();
+
+                    log.info("device#capture#_makeCaptureOption():" + " dir name : " + parseDirPath 
+                        + " ,file name : " + parseBase + " ,inputFormat : " + parseExt);
+
+                    // if given path has extension
+                    if (parseBase && parseExt) {
+                        if (parseExt === "png" || parseExt === "bmp" || parseExt === "jpg") {
+                            captureFormat = parseExt.toUpperCase();
```

Comment:
why this convert to uppercase?

Reply:
Need to set "format" as uppercase. 

# luna-send -n 1 luna://com.webos.surfacemanager/captureCompositorOutput '{"output":"/tmp/settings.bmp","format":"bmp"}'
{
    "errorCode": 104,
    "errorText": "ERR_INVALID_FORMAT",
    "returnValue": false
}